### PR TITLE
Update ObjFW definition

### PR DESCRIPTION
**Pull request by Frityet**
This updates the ObjFW definition in order to be more accurate and work better

### CODE REVIEW

File: packages/o/objfw/xmake.lua [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package("objfw")
+package("objfw-local")
```
Comment (waruqi) [inline packages/o/objfw/xmake.lua:1]: why?

**Comment by waruqi** [comment]
please push to dev branch next

File: packages/o/objfw/xmake.lua [hunk 2]
```diff
@@ -12,13 +12,47 @@ package("objfw")
         add_frameworks("CoreFoundation")
     end
 
-    on_install("linux", "macosx", function (package)
-        local configs = {"--without-tls"}
+    add_configs("tls", {
```
Comment (waruqi) [inline packages/o/objfw/xmake.lua:15]: please add it in single line.

File: packages/o/objfw/xmake.lua [hunk 3]
```diff
@@ -12,13 +12,47 @@ package("objfw")
...
+        local configs = {}
+        table.insert(configs, (package:config("tls") and "" or "--without-tls"))
         table.insert(configs, "--enable-shared=" .. (package:config("shared") and "yes" or "no"))
-        if package:debug() then
+        if package:is_debug() then
             table.insert(configs, "--enable-debug")
         end
         import("package.tools.autoconf").install(package, configs)
+
+        local tool = import("lib.detect.find_tool")
```
Comment (waruqi) [inline packages/o/objfw/xmake.lua:36]: remove `local tool = ` and use `find_tool`

File: packages/o/objfw/xmake.lua [hunk 4]
```diff
@@ -12,13 +12,47 @@ package("objfw")
...
+
+        local objfwcfg = tool("objfw-config", { paths = { package:installdir("bin") } }).program
```
Comment (waruqi) [inline packages/o/objfw/xmake.lua:38]: find_tool

File: packages/o/objfw/xmake.lua (hunk 1 above)
Comment (Frityet) [reply packages/o/objfw/xmake.lua:1]: Whoops!

**Comment by Frityet** [comment]
> please push to dev branch next

Will do, ill make a new PR

File: packages/o/objfw/xmake.lua (hunk 4 above)
Comment (Frityet) [reply packages/o/objfw/xmake.lua:38]: got it

File: packages/o/objfw/xmake.lua (hunk 3 above)
Comment (Frityet) [reply packages/o/objfw/xmake.lua:36]: kk

**Pull request closed without merging**